nxc/protocols/winrm.py

```diff
@@ -207,27 +230,39 @@ def create_conn_obj(self):
                 else:
                     self.logger.info(f"Other ConnectionError to WinRM service: {e}")
         return False
-
+    
+    def check_if_admin(self):
+        wsman = self.conn.wsman
+        wsen = NAMESPACES["wsen"]
+        wsmn = NAMESPACES["wsman"]
+
+        enum_msg = ET.Element(f"{{{wsen}}}Enumerate")
+        ET.SubElement(enum_msg, f"{{{wsmn}}}OptimizeEnumeration")
+        ET.SubElement(enum_msg, f"{{{wsmn}}}MaxElements").text = "32000"
+
+        wsman.enumerate("http://schemas.microsoft.com/wbem/wsman/1/windows/shell", enum_msg)
+        self.admin_privs = True
+        return True
+        
     def plaintext_login(self, domain, username, password):
+        self.admin_privs = False
+        if not self.args.laps:
+            self.password = password
+            self.username = username
+        self.domain = domain
+
         try:
-            if not self.args.laps:
-                self.password = password
-                self.username = username
-            self.domain = domain
             self.conn = Client(
                 self.host,
                 auth="ntlm",
-                username=f"{domain}\\{self.username}",
+                username=self.username if self.local_auth else f"{self.username}@{self.domain.upper()}",
```

Comment:
I don't like this local-auth fix https://github.com/Pennyw0rth/NetExec/pull/72/commits/415cd1afa0b07306cac929868462464e8c459fcf

it's way overkill while the most simple solution is just `username=f"{self.domain}\\{self.username}"` 

Any reason to go with a new logic for local auth regarding all the other protocols ?

Reply:
I ask the author of pypsrp, it only support upn format

Reply:
You can test with develop branch, when you given a non-existed domain with -d, it still get own flag

Reply:
https://github.com/jborean93/pypsrp/issues/180

Reply:
I think you misinterpret the result, on the develop branch you see the pwn3d message because the login is successful and the user can execute command. Since it's local auth, it doesn't matter what domain you are using --local-auth or -d yolo is the same.

You are trying to fix a problem related to kebreros which is not the subject of the PR :)

Reply:
> I think you misinterpret the result, on the develop branch you see the pwn3d message because the login is successful and the user can execute command. Since it's local auth, it doesn't matter what domain you are using --local-auth or -d yolo is the same.

AH, yes, little bit confused about this XD